### PR TITLE
MultiQC docs migration

### PR DESCRIPTION
**Pull request by llewellyn-sl**

### CODE REVIEW

File: multiqc_docs/sidebar.json [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+      "label": "Supported tools",
+      "collapsed": true,
+      "items": [
+        "modules/adapterremoval",
+        "modules/afterqc",
+        "modules/anglerfish",
+        "modules/bakta",
+        "modules/bamdst",
+        "modules/bamtools",
+        "modulest/bbduk",
```
Comment (ewels) [inline multiqc_docs/sidebar.json:61]:
```suggestion
        "modules/bbduk",
```

File: multiqc_docs/sidebar.json [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        "core/getting_started/config"
+      ]
+    },
+    {
+      "type": "category",
+      "label": "Reports",
+      "collapsed": true,
+      "items": [
+        "core/reports/reports",
+        "core/reports/customization"
```
Comment (ewels) [inline multiqc_docs/sidebar.json:21]: 🇬🇧 
```suggestion
        "core/reports/customisation"
```

**Review by ewels — COMMENTED** [review]
Hmm, annoying that every markdown file needs to be listed explicitly like this.. Means that we can't just edit the MultiQC repo and expect the Seqera docs site to reflect changes necessarily.

It'd be nicer if we could just glob the files somehow with a pattern (eg. `*md` or whatever). Maybe something to look into in the future.

**Comment by jason-seqera** [comment]
@ewels, so I _think_ this can be a JavaScript file, so you could theoretically then output the necessary JSON structure from that. Would have to look at the sidebar API for Docusaurus to be sure.

**Comment by ewels** [comment]
@jason-seqera - you're absolutely right. Got it working, much better ✨ 

Thanks for the tip!

**Comment by ewels** [comment]
It's getting late, so don't want to do it now. But note that with the autogenerated sidebar we can also put the sidebar position in the markdown frontmatter (see [docs](https://docusaurus.io/docs/sidebar/autogenerated#autogenerated-sidebar-metadata)). Then the `sidebar.js` becomes very thin and basically everything is in the markdown files themselves.

I think that this will be a lot easier, as then almost nothing needs to be touched in the docs repo for updates. Can add, remove and reorder docs pages directly from the markdown in the code repos themselves rather than being split between two and needing to time PR merges precisely.

**Comment by jason-seqera** [comment]
@ewels, I don't like that declarative navigation approach because it's either alphabetical or you need to set the logical position of every single page, and keep track of it by flipping through multiple files at times. It's scattered. And if you mix and match declarative and omitting, I think you get some weird undefined behavior where ordered stuff shows up inter-spliced with alphabetical. It felt unworkable if you want fine-grained control. And you need run a preview to be sure exactly where a page is going to appear, rather than just explicitly declaring it in relation to all other pages in the sidebar data structure itself.

So in my testing I ultimately wasn't a fan.

**Comment by ewels** [comment]
Fair enough 👍🏻 I was thinking that the main downside with the current approach is that we have to be a bit careful with (a) removing or renaming any markdown files in the code repo, as that will break the docs build. And (b) remembering to open a second PR to this repo if we ever add a new markdown file. But it's probably not a big effort / problem in this use case.

I'm pretty used to using front matter sort order / weights in markdown and have been doing that for years, so it also feels fairly natural for me. We do it on the rather large docs base in nf-core, for example.

As long as we have the automated glob for modules, as above ☝🏻 then I'm happy.

**Comment by llewellyn-sl** [comment]
@ewels other than the extra content from the modules pages, is this PR good to go? Do we need to do edits on the MultiQC side to pull into here before we merge?

**Comment by ewels** [comment]
> Do we need to do edits on the MultiQC side to pull into here before we merge?

No, we can host this off the `overview-svg-inline` branch for now and merge with that. So I'm thinking the workflow can be:

1. Remove the [multiqc.info](http://multiqc.info/) docs
    * Update the MultiQC website to link to [docs.seqera.io](http://docs.seqera.io/) from the navbar
    * Set up redirects on [multiqc.info](http://multiqc.info/) so that old links go to the new docs
    * Deprecate + delete the docs part of the MultiQC website
2. Merge the [docs refactoring PR](https://github.com/MultiQC/MultiQC/pull/2643) into MultiQC `main`
3. Update this docs.seqera.io repo to build off `main` instead (and then delete `overview-svg-inline`)
4. Write MultiQC automation to copy the search patterns into the supported tool docs

> is this PR good to go

Want to do a bit more checking. Just noticed two things for example:

1. Links / sidebar highlight colour is Wave, not MultiQC
2. There are some broken internal links, e.g. _"see full docs"_ [here](https://deploy-preview-84--seqera-docs.netlify.app/multiqc/getting_started/quick_start)

I'd like to do more checking for broken links etc if possible.. Not sure if there's a way to automate that?

**Comment by jason-seqera** [comment]
@ewels, broken links show up if you run a `build` for Docusaurus rather than live preview. there are also these: https://github.com/untitaker/hyperlink#alternatives

**Comment by ewels** [comment]
Be careful what you wish for - there are quite a lot of warnings about broken links 😆 

<details>

Exhaustive list of all broken links found:

- On source page path = `/multiqc/getting_started/quick_start`:
   - -> linking to `../installation#python-with-conda` (resolved as: `/multiqc/installation`)
   - -> linking to `../installation#conda` (resolved as: `/multiqc/installation`)

- On source page path = `/multiqc/getting_started/running_multiqc`:
   - -> linking to `../reports/reports#exporting-plots` (resolved as: `/multiqc/reports/reports`)

- On source page path = `/platform/22.4.0/compute-envs/aws-batch`:
   - -> linking to `../credentials/overview.mdx#container-registry-credentials` (resolved as: `/platform/22.4.0/credentials/overview.mdx`)

- On source page path = `/platform/22.4.0/compute-envs/azure-batch`:
   - -> linking to `../credentials/overview.mdx#container-registry-credentials` (resolved as: `/platform/22.4.0/credentials/overview.mdx`)

- On source page path = `/platform/22.4.0/compute-envs/eks`:
   - -> linking to `../k8s#cluster-preparation` (resolved as: `/platform/22.4.0/k8s`)
   - -> linking to `../credentials/overview.mdx#container-registry-credentials` (resolved as: `/platform/22.4.0/credentials/overview.mdx`)

- On source page path = `/platform/22.4.0/compute-envs/gke`:
   - -> linking to `../google-cloud#configure-google-cloud` (resolved as: `/platform/22.4.0/google-cloud`)
   - -> linking to `../k8s#cluster-preparation` (resolved as: `/platform/22.4.0/k8s`)
   - -> linking to `../credentials/overview.mdx#container-registry-credentials.mdx` (resolved as: `/platform/22.4.0/credentials/overview.mdx`)

- On source page path = `/platform/22.4.0/compute-envs/google-cloud-lifesciences`:
   - -> linking to `../credentials/overview.mdx#container-registry-credentials` (resolved as: `/platform/22.4.0/credentials/overview.mdx`)

- On source page path = `/platform/22.4.0/compute-envs/k8s`:
   - -> linking to `../eks` (resolved as: `/platform/22.4.0/eks`)
   - -> linking to `../gke` (resolved as: `/platform/22.4.0/gke`)

- On source page path = `/platform/22.4.0/orgs-and-teams/workspace-management`:
   - -> linking to `/docs/limits/limits.mdx

- On source page path = `/platform/22.4.0/resource-labels/overview`:
   - -> linking to `/docs/_templates/aws-batch/forge-policy.json

- On source page path = `/platform/22.4.0/supported_software/fusion`:
   - -> linking to `/docs/compute-envs/aws-batch.mdx
   - -> linking to `/docs/compute-envs/aws-batch.mdx#` (resolved as: `/docs/compute-envs/aws-batch.mdx`)

- On source page path = `/platform/23.1.0/enterprise`:
   - -> linking to `./advanced-topics/seqera-container-images.mdx` (resolved as: `/platform/23.1.0/advanced-topics/seqera-container-images.mdx`)

- On source page path = `/platform/23.1.0/enterprise/prerequisites/aws`:
   - -> linking to `../advanced-topics/seqera-container-images.mdx` (resolved as: `/platform/23.1.0/enterprise/advanced-topics/seqera-container-images.mdx`)

- On source page path = `/platform/23.1.0/enterprise/prerequisites/azure`:
   - -> linking to `../advanced-topics/seqera-container-images.mdx` (resolved as: `/platform/23.1.0/enterprise/advanced-topics/seqera-container-images.mdx`)

- On source page path = `/platform/23.1.0/enterprise/prerequisites/gcp`:
   - -> linking to `../advanced-topics/seqera-container-images.mdx` (resolved as: `/platform/23.1.0/enterprise/advanced-topics/seqera-container-images.mdx`)

- On source page path = `/platform/23.1.0/enterprise/release_notes/changelog`:
   - -> linking to `6d98210c` (resolved as: `/platform/23.1.0/enterprise/release_notes/6d98210c`)

- On source page path = `/platform/23.1.0/orgs-and-teams/workspace-management`:
   - -> linking to `/docs/limits/limits.mdx

- On source page path = `/platform/23.2.0/credentials/agent_credentials`:
   - -> linking to `../agent.mdx` (resolved as: `/platform/23.2.0/agent.mdx`)

- On source page path = `/platform/23.2.0/enterprise`:
   - -> linking to `./advanced-topics/seqera-container-images.mdx` (resolved as: `/platform/23.2.0/advanced-topics/seqera-container-images.mdx`)

- On source page path = `/platform/23.2.0/enterprise/advanced-topics/manual-aws-batch-setup`:
   - -> linking to `../../compute-envs/aws-batch.md#tower-forge-compute-environment` (resolved as: `/platform/23.2.0/compute-envs/aws-batch.md`)
   - -> linking to `../../compute-envs/aws-batch.md#manual` (resolved as: `/platform/23.2.0/compute-envs/aws-batch.md`)

- On source page path = `/platform/23.2.0/enterprise/prerequisites/aws`:
   - -> linking to `../advanced-topics/seqera-container-images.mdx` (resolved as: `/platform/23.2.0/enterprise/advanced-topics/seqera-container-images.mdx`)

- On source page path = `/platform/23.2.0/enterprise/prerequisites/azure`:
   - -> linking to `../advanced-topics/seqera-container-images.mdx` (resolved as: `/platform/23.2.0/enterprise/advanced-topics/seqera-container-images.mdx`)

- On source page path = `/platform/23.2.0/enterprise/prerequisites/gcp`:
   - -> linking to `../advanced-topics/seqera-container-images.mdx` (resolved as: `/platform/23.2.0/enterprise/advanced-topics/seqera-container-images.mdx`)

- On source page path = `/platform/23.2.0/getting-started/usage`:
   - -> linking to `../cli.mdx` (resolved as: `/platform/23.2.0/cli.mdx`)

- On source page path = `/platform/23.2.0/orgs-and-teams/workspace-management`:
   - -> linking to `/docs/limits/limits.mdx

- On source page path = `/platform/23.3.0/enterprise/configuration/overview`:
   - -> linking to `../../administration/audit-logs.mdx` (resolved as: `/platform/23.3.0/administration/audit-logs.mdx`)

- On source page path = `/platform/23.3.0/enterprise/release_notes/changelog`:
   - -> linking to `6d98210c` (resolved as: `/platform/23.3.0/enterprise/release_notes/6d98210c`)

- On source page path = `/platform/23.3.0/orgs-and-teams/workspace-management`:
   - -> linking to `../orgs-and-teams.shared-workspaces.mdx` (resolved as: `/platform/23.3.0/orgs-and-teams.shared-workspaces.mdx`)
   - -> linking to `/docs/limits/limits.mdx

- On source page path = `/platform/23.4.0/enterprise/configuration/overview`:
   - -> linking to `../../administration/audit-logs.mdx` (resolved as: `/platform/23.4.0/administration/audit-logs.mdx`)

- On source page path = `/platform/23.4.0/enterprise/release_notes/changelog`:
   - -> linking to `6d98210c` (resolved as: `/platform/23.4.0/enterprise/release_notes/6d98210c`)

</details>


Most are Platform docs and I'll ignore. I can clean up the 2 MultiQC ones. Nice to know about this, thanks @jason-seqera!

**Review by ewels — APPROVED** [review]
Ok, I can't think of anything else left to do before merging ✅

**Review by ewels — CHANGES_REQUESTED** [review]
Belay that - please hold until new design work discussed with @deepcshah 

Will get back with updates ASAP.

**Comment by ewels** [comment]
Note for self: it would be nice not to have the huuuge sidebar dropdown with all tools labelled.

Instead, have a single link to a MDX index page for it, and have a React MDX component of some kind to show a filterable table. Same as https://multiqc.info/modules/

**Review by ewels — APPROVED** [review]
Ok, so plan after quick chat just now:

1. Merge this PR without design changes, as-is
2. @deepcshah to work on getting a new design in Figma and approve (aim by end of next week)
3. @MythicalFish to put in a new PR with design updates as time allows

So @llewellyn-sl happy for you to merge if you're ok with the docs as they stand in this PR 👍🏻

**Pull request merged**